test/e2e/framework/service/jig.go

```diff
@@ -82,6 +111,68 @@ func NewTestJig(client clientset.Interface, namespace, name string) *TestJig {
 	return j
 }
 
+// allocatePort reserves the port provided as input.
+// If an invalid port was provided or if the port is already reserved, it returns false
+func (s *staticPortRange) allocatePort(port int) bool {
+	s.Lock()
+	defer s.Unlock()
+	port -= s.baseport
+	if port < 0 || port > s.length || s.reservedPorts.Has(port) {
+		return false
+	}
+	s.reservedPorts.Insert(port)
+	return true
+}
+
+// nextFreePort returns a free port from the range and returns its number and true
+// the port is not allocated so the consumer should allocate it explicitly calling allocatePort()
+// if none is available then it returns -1 and false
+func (s *staticPortRange) nextFreePort() (int, bool) {
+	s.Lock()
+	defer s.Unlock()
+	// start in a random offset
+	start := rand.Intn(s.length)
+	for i := 0; i < s.length; i++ {
+		port := (start + i) % s.length
+		if !s.reservedPorts.Has(port) {
+			return s.baseport + port, true
+		}
+	}
+	return -1, false
+}
+
+// release the port passed as an argument and returns true. If
+// an invalid port or unreserved port was provided, it returns false
+func (s *staticPortRange) release(port int) bool {
+	s.Lock()
+	defer s.Unlock()
+	port -= s.baseport
+	// port is out of range, it can not be released
+	if port < 0 || port > s.length || !s.reservedPorts.Has(port) {
+		return false
+	}
+	s.reservedPorts.Delete(port)
+	return true
+}
+
+// ReserveStaticNodePort reserves the port provided as input.
+// If an invalid port was provided or if the port is already reserved, it returns false
+func ReserveStaticNodePort(port int) bool {
+	return staticPortAllocator.allocatePort(port)
+}
+
+// GetUnusedStaticNodePort returns a free port in static range and a boolean True value
+// If no port in static range is available it returns -1 and a boolean value False
+func GetUnusedStaticNodePort() (int, bool) {
+	return staticPortAllocator.nextFreePort()
+}
+
+// ReleaseStaticNodePort releases a previously reserved static port and returns true
+// If the port was not previously reserved it returns false
+func ReleaseStaticNodePort(port int) bool {
+	return staticPortAllocator.release(port)
+}
+
```

Comment:
same as https://github.com/kubernetes/kubernetes/pull/127153/files#r1812469730 we never check the output of the operation and there is no action we can take
```suggestion
// ReleaseStaticNodePort releases the specified port
func ReleaseStaticNodePort(port int) {
	return staticPortAllocator.release(port)
}

```